---
title: "R Notebook"
output: html_notebook
---

Human TFAP2A OE RNA-Seq Analysis

```{r}
library(reshape2)
library(ggplot2)
library(ggthemes)
library(ggrepel)
library(dplyr)
library(ggsci)
library(ggpubr)
library(readr)
library(stringr)
library(DESeq2)
library(BiocParallel)
library(GenomicAlignments)
library(org.Hs.eg.db)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(rtracklayer)
library(ChIPQC)
library(ChIPseeker)
library(ATACseqQC)
TxDb_hg38 <- TxDb.Hsapiens.UCSC.hg38.knownGene
library(AnnotationHub)
ah <- AnnotationHub()
qr <- query(ah, c("EnsDb", "Homo sapiens"))
ensembl_human_annotation <- qr[["AH95744"]]
register(MulticoreParam(32)) # Set this to number of CPUs.

GenesUniverse <- read_csv(file = "Imports/GenesUniverse.csv")
GenesUniverse$Human_ENSEMBL <- mapIds(org.Hs.eg.db, keys = as.character(GenesUniverse$Human_Entrezid),
                                      keytype = "ENTREZID", column = "ENSEMBL", multiVals = "first")
GenesUniverse$Human_ENSEMBL <- as.character(GenesUniverse$Human_ENSEMBL)
GenesUniverse$Human_ENSEMBL[which(GenesUniverse$Human_ENSEMBL == "NULL")] <- NA
GenesUniverse <- as.data.frame(GenesUniverse)
```

```{r}
featureCounts <- read.delim("~/local_git/NC_Timecourse/RNA-Seq/Imports/TFAP2A-OE2_featureCounts.txt", comment.char="#")

#Remove the gene information to just get the counts matrix.
dat <- featureCounts[,7:10]
rownames(dat) <- featureCounts$Geneid

sample_names <- colnames(featureCounts[7:10])
sample_names <- str_sub(sample_names, 7,-5)
sample_names <- word(sample_names, sep = "_", 5, 8)

colnames(dat) <- sample_names

colData <- data.frame(row.names=colnames(dat), data=colnames(dat),
                      Condition = c("GFP_OE", "GFP_OE",
                                    "AP2A_OE","AP2A_OE"),
                      Replicate = c(1,2,1,2))

dds <- DESeqDataSetFromMatrix(countData = dat,
                              colData = colData,
                              design= ~Condition)
#Filter out genes lowly detected.
keep <- rowSums(counts(dds)) >= 15
dds <- dds[keep,]


dds <- DESeq(dds, parallel = TRUE)

#Check out the counts
bp1 <-boxplot(log10(counts(dds,normalized=TRUE)+1))

vstDds <- varianceStabilizingTransformation(dds)
pcaAll <- plotPCA(vstDds, intgroup = c("Condition"), returnData=TRUE)
pcaAllVar <- attr(pcaAll, "percentVar")

g <- ggplot(pcaAll, aes(x=PC1, y=PC2, color=Condition)) + 
  geom_point(size = 4, alpha = .75) + 
  scale_color_npg() +
  ggtitle("pcaALL") +
  labs(x = paste(c("PC1","%"),round(pcaAllVar[1]*100)) ,y = paste(c("PC2","%"), round(pcaAllVar[2]*100))) +
  theme(text=element_text(color="black", size = 10, face = "bold"),
        line=element_line(color="black", size = 2),
        axis.ticks = element_line(color = "black", size = 1),
        panel.background = element_blank(),
        panel.border = element_rect(fill=NA, color = "black", size = 2),
        axis.text = element_text(color="black", size = 10, face = "bold")
        #legend.key = element_blank()
  )
g
ggsave(filename = "Exports/TFAP2A_OE_RNA_PCA.png")
g

```

```{r}
restest <- as.data.frame(results(dds, name = "Condition_GFP_OE_vs_AP2A_OE", parallel = TRUE))
restest$Symbol <- mapIds(org.Hs.eg.db, keys = rownames(restest), keytype = "ENSEMBL", column = "SYMBOL")
restest$log2FoldChange <- restest$log2FoldChange *-1

TFAP2A_OE_sig <- restest[restest$log2FoldChange > 1 & restest$pvalue < 0.05,]
Decreased_Expression <- restest[restest$log2FoldChange < -1 & restest$pvalue < 0.05,]

write_csv(restest, file = "./Exports/TFAP2A_OE2_Unfiltered.csv")

positive_factors <- c("DLX1","COL3A1","CDX2","DLX2","ITGB6","S100A9","TWIST1","MIXL1","CDX1","MSX1","TFAP2A","GATA6","WNT9B","WNT6","GATA3","WNT5A","NR2F1","SNAI2","TFAP2B","SP5")
negative_factors <- c("SIX3","LHX2","OLIG3","FEZF1","SOX21","SIX6","HESX1")

restest_title <- "TFA2A_OE"

restest3 <- restest %>% mutate(threshold = ifelse(log2FoldChange >= 1.5 & padj < 0.05, "A", ifelse(log2FoldChange <= -1.5 & padj < 0.05, "B", "C")))

g <- ggplot(data = restest3, aes(x = log2FoldChange, y = -log10(padj))) +
        geom_point(aes(color = threshold), size = 1, alpha = 1) +
        scale_color_manual(values = c("A" = "#e45649", "B" = "#0584bc", "C" = "#d3d2d6")) +
        geom_text_repel(data=restest3[restest3$Symbol %in% c(positive_factors, negative_factors),],
                        aes(label = Symbol), box.padding = 0.2, segment.color = "#393a42", segment.size = 0.5, color = "#393a42") +
        ggtitle(restest_title) +
        #scale_x_continuous(limits = c(-6,6)) +
        geom_vline(xintercept = c(-1.5,1.5), linetype = "longdash", color = "black", size = 0.4) +
        theme_classic() +
        xlab("Log2 Fold Change") +
        ylab(paste0("-Log10 Adj. p value, 0.001 Labeled")) +
        theme(legend.position = "none") +
        scale_y_continuous(trans = "pseudo_log")
        # , axis.title = element_blank(),
        #      axis.text = element_blank(),
        #      axis.ticks = element_blank(),
        #      plot.title = element_blank()) #plot.title = element_text(hjust = 0.5))
g
      

ggsave(filename = paste0("./Exports/",restest_title,"_labeled.png"), plot = g, width = 5, height = 4, device = "png")
ggsave(filename = paste0("./Exports/",restest_title,"_labeled.pdf"), plot = g, width = 5, height = 4, device = "pdf")

```

Let's investigate how the NC GRN looks compared to random 250 genes.
```{r}
GRN_Genes <- read_delim(file = "/home/ash274/local_git/NC_Timecourse/ATAC-Seq/Imports/GRN_Genes.tsv",
                      col_names = F, delim = "    ")

colnames(GRN_Genes) <- c("Symbol", "Chick_ENSEMBL")
GRN_Genes$GRN <- "YES"
GenesUniverse_GRN <- merge(GenesUniverse, GRN_Genes, by = "Chick_ENSEMBL", all = T)
GenesUniverse_GRN$GRN <- !is.na(GenesUniverse_GRN$GRN)


grn_ens <- as.character(GenesUniverse_GRN[GenesUniverse_GRN$GRN,c("Human_ENSEMBL")])
grn_ens <- grn_ens[!grn_ens == "NA"]
grn_ens <- grn_ens[!grn_ens == "NULL"]
grn_ens <- grn_ens[grn_ens %in% rownames(restest)]

othertf_ens <- GenesUniverse_GRN[!GenesUniverse_GRN$GRN & GenesUniverse_GRN$TF_Status == "Transcription Factor",c("Human_ENSEMBL")]
othertf_ens <- as.character(othertf_ens)
othertf_ens <- othertf_ens[!othertf_ens == "NA"]
othertf_ens <- othertf_ens[!othertf_ens == "NULL"]
othertf_ens <- othertf_ens[othertf_ens %in% rownames(restest)]

restest$In_GRN <- ifelse(rownames(restest) %in% grn_ens, "IN_GRN",
                         ifelse(rownames(restest) %in% othertf_ens,"OTHER_TF", "NO"))
restest_sub <- restest[restest$In_GRN %in% c("IN_GRN", "OTHER_TF"),]
restest_sub_GRN <- restest[restest$In_GRN %in% c("IN_GRN"),]

ggplot(data = restest_sub, mapping = aes(x = In_GRN, y = log2FoldChange)) + 
  geom_violin() + stat_compare_means(method = "t.test")
```

## Compare against RNA-Seq Timecourse
Thanks Jackie for this data!
```{r}
library(readr)
library(stringr)
timecourse_featurecounts <- read_delim(file = "/data/Jackie/NCinduction_TimeCourse/RNA_TC_human/featureCounts.txt", delim = "\t", comment = "#")
rownames(timecourse_featurecounts) <- timecourse_featurecounts$Geneid
timecourse_metadata <- read_csv("/data/Jackie/NCinduction_TimeCourse/RNA_TC_human/RNA_seq_metadata.csv")

timecourse_dat <- as.data.frame(timecourse_featurecounts[,7:18])
rownames(timecourse_dat) <- timecourse_featurecounts$Geneid
colnames(timecourse_dat) <- c("D3_2","D4_2","D5_2","D0_1",
                              "D1_1","D2_1","D3_1","D4_1",
                              "D5_1","D0_2","D1_2","D2_2")

timecourse_meta2 <- data.frame(data = colnames(timecourse_dat),
                               Condition = c("D3","D4","D5","D0",
                                             "D1","D2","D3","D4",
                                             "D5","D0","D1","D2"),
                               Replicate = c("2","2","2","1",
                                             "1","1","1","1",
                                             "1","2","2","2"))

featureCounts <- read.delim("~/local_git/NC_Timecourse/RNA-Seq/Imports/TFAP2A-OE2_featureCounts.txt", comment.char="#")

#Remove the gene information to just get the counts matrix.
dat <- featureCounts[,7:10]
rownames(dat) <- featureCounts$Geneid

sample_names <- colnames(featureCounts[7:10])
sample_names <- str_sub(sample_names, 7,-5)
sample_names <- word(sample_names, sep = "_", 5, 8)

colnames(dat) <- sample_names

colData <- data.frame(row.names=colnames(dat), data=colnames(dat),
                      Condition = c("GFP_OE", "GFP_OE",
                                    "AP2A_OE","AP2A_OE"),
                      Replicate = c(1,2,1,2))

combined_metadata <- rbind(timecourse_meta2, colData)
# filter dat to match those in timecourse_dat
dat_filt <- dat[rownames(dat) %in% rownames(timecourse_dat),]
summary(rownames(dat_filt) == rownames(timecourse_dat))
combined_dat <- cbind(timecourse_dat, dat_filt)

# Now make D0 and either D3 or D5.
combined_dat <- combined_dat[,c("D0_1","D0_2","D5_1","D5_2",
                                "HGHM3BGXK_H1_GFP_RNAs1","HGHM3BGXK_H1_GFP_RNAs2",
                                "HGHM3BGXK_H1_Ap2A_RNAs1","HGHM3BGXK_H1_Ap2A_RNAs2")]

combined_metadata <- combined_metadata[combined_metadata$Condition %in% c("D0","D5","GFP_OE","AP2A_OE"),]

# For D0/3/5
# combined_metadata <- combined_metadata[c(3,6,4,1,5,2,7,8,9,10),]

# For D0/3
# combined_metadata <- combined_metadata[c(2,4,3,1,5,6,7,8),]

# For D0/5
combined_metadata <- combined_metadata[c(2,4,3,1,5,6,7,8),]

combined_metadata$data == colnames(combined_dat)

combined_dds <- DESeqDataSetFromMatrix(countData = combined_dat,
                              colData = combined_metadata,
                              design= ~Condition + 0)
#Filter out genes lowly detected.
keep <- rowSums(counts(combined_dds)) >= 10
combined_dds <- combined_dds[keep,]


combined_dds <- DESeq(combined_dds, parallel = TRUE)

#Check out the counts
bp1 <-boxplot(log10(counts(combined_dds,normalized=TRUE)+1))

combined_dds_rlog <- rlog( combined_dds, blind = T )
rv <- rowVars(assay(combined_dds_rlog))
# select the ntop genes by variance
select <- order(rv, decreasing=TRUE)[seq_len(min(750, length(rv)))]
# perform a PCA on the data in assay(x) for the selected genes
pca <- prcomp(t(assay(combined_dds_rlog)[select,]))
pca_df <- as.data.frame(pca$x)
pca_df$Condition <- combined_dds_rlog$Condition

g <- ggplot(pca_df, aes(x=PC2, y=PC3, color=Condition)) + 
  geom_point(size = 4, alpha = .75) + 
  scale_color_npg() +
  ggtitle("pcaALL") +
  labs(x = c("PC2 14.2%"),y = c("PC3 9.2%")) +
  theme(text=element_text(color="black", size = 10, face = "bold"),
        line=element_line(color="black", size = 2),
        axis.ticks = element_line(color = "black", size = 1),
        panel.background = element_blank(),
        panel.border = element_rect(fill=NA, color = "black", size = 2),
        axis.text = element_text(color="black", size = 10, face = "bold")
        #legend.key = element_blank()
  )
g
ggsave("Exports/TFAP2A_OE_RNA_PCA_D0-D5.png", width = 6, height = 4)
ggsave("Exports/TFAP2A_OE_RNA_PCA_D0-D5.pdf", width = 6, height = 4)

# Highest Discriminating PC3 Genes
PC3 <- as.data.frame(pca$rotation[,3])
colnames(PC3) <- "value"
PC3$Human_ENSEMBL <- rownames(PC3)
top_100_PC3 <- PC3 %>% slice_max(value, n = 100)
top_100_PC3 <- merge(top_100_PC3, GenesUniverse, all.x = T)
```
Let's compare the enrichment of TFAP2A-OE genes within each of these samples.

```{r}
library(ggpubr)

TFAP2A_OE_Genes <- restest %>% dplyr::filter(log2FoldChange > 1.5 & padj < 0.01)

# 1000 Random not significant genes
Random_1000 <- restest %>% dplyr::filter(padj > 0.05) %>% slice_sample(n = 1000)

rlog_counts <- assay(combined_dds_rlog)
colnames(rlog_counts) <- colData(combined_dds_rlog)$data

rlog_counts_melted <- melt(rlog_counts)
rlog_counts_melted$Group <- ifelse(rlog_counts_melted$Var1 %in% rownames(TFAP2A_OE_Genes),
                                   "TFAP2A-OE_Enriched",
                                   ifelse(rlog_counts_melted$Var1 %in% rownames(Random_1000),"Control",
                                          "Other"))
rlog_counts_melted$Factor <- rlog_counts_melted$Var2
levels(rlog_counts_melted$Factor) <- c("D0","D0","D5","D5","GFP-OE","GFP-OE","TFAP2A-OE","TFAP2A-OE")
rlog_counts_melted$Factor <- relevel(rlog_counts_melted$Factor, ref = "GFP-OE")
rlog_counts_melted$Factor <- relevel(rlog_counts_melted$Factor, ref = "TFAP2A-OE")


my_comparisons <- list( c("D0", "D5"), c("D0", "GFP-OE"), c("TFAP2A-OE","D5"), c("TFAP2A-OE","GFP-OE")
                        , c("D5","GFP-OE"))

ggplot(rlog_counts_melted %>% dplyr::filter(Group == "TFAP2A-OE_Enriched"), aes(Factor, value)) + geom_boxplot(aes(fill = Factor)) + theme(axis.text.x = element_blank()) + stat_compare_means(comparisons = my_comparisons, method.args = list(alternative = "two.sided")) + scale_fill_brewer(palette = "Set1") + ggtitle("Enrichment of TFAP2A-OE induced gene expression as compared to wild type.") + xlab("RNA-Seq Library") + ylab("Rlog Counts")

ggsave(file = "Exports/TFAP2A_OE_Comparison_D5.png", width = 6, height = 3.5)
ggsave(file = "Exports/TFAP2A_OE_Comparison_D5.pdf", width = 6, height = 3.5)

```

Integrate TFAP2A-OE ATAC samples.
```{r}
OE_dba <- dba(sampleSheet="../ATAC-Seq/Human_ESC_iNC/TFAP2A_OE_atATAC_samplesheet.csv")
OE_dba <- dba.count(OE_dba, bParallel = T, summits = F)
OE_dba <- dba.contrast(OE_dba, group1 = OE_dba$masks$`TFAP2A-OE`, group2= OE_dba$masks$`GFP-OE`, minMembers = 2)
OE_dba <- dba.contrast(OE_dba, group1 = OE_dba$masks$ES, group2= OE_dba$masks$NC, minMembers = 2)
OE_dba <- dba.analyze(OE_dba, bParallel = T)
GFP_vs_TFAP2A <- as.data.frame(dba.report(OE_dba, contrast = 1, bUsePval = T))
GFP_vs_TFAP2A$PeakID <- paste0(GFP_vs_TFAP2A$seqnames, "_", GFP_vs_TFAP2A$start, "_", GFP_vs_TFAP2A$end)

write.csv(GFP_vs_TFAP2A, file = "../ATAC-Seq/Human_ESC_iNC/Exports/GFP_vs_TFAP2A.csv", row.names = F)

TFAP2A_Enriched <- GFP_vs_TFAP2A[GFP_vs_TFAP2A$Fold > 1.5,]
#TFAP2A_Enriched$seqnames <- stringr::str_split(TFAP2A_Enriched$seqnames, pattern = "chr", simplify = T)[,2]
TFAP2A_Enriched$strand <- "+"
write_tsv(TFAP2A_Enriched[,c("seqnames","start","end","PeakID","Fold","strand")],
          file = "../ATAC-Seq/Human_ESC_iNC/Exports/TFAP2A_OE_Enriched_Peaks.bed",
          col_names = F)

background_peaks <- as.data.frame(dba.report(OE_dba, contrast = 1, bUsePval = T, th = 0.4))
background_peaks$strand <- "+"
background_peaks$PeakID <- paste0(background_peaks$seqnames, "_", background_peaks$start, "_", background_peaks$end)
background_peaks <- background_peaks[background_peaks$p.value > 0.1,]
write_tsv(background_peaks[,c("seqnames","start","end","PeakID","Fold","strand")],
          file = "../ATAC-Seq/Human_ESC_iNC/Exports/Background_Peaks.bed",
          col_names = F)



meta_dataframe <- as.data.frame(OE_dba$class)

pca <- dba.plotPCA(OE_dba, attributes = c(DBA_TREATMENT, DBA_CONDITION), bLog = F)
x <- pca$panel.args[[1]]$x
y <- pca$panel.args[[1]]$y
pca_mat <- as.data.frame(matrix(c(x,y), nrow = 8, ncol = 2, byrow = F))
colnames(pca_mat) <- c("PC1","PC2")
rownames(pca_mat) <- OE_dba$samples$SampleID

pca_mat$condition <- factor(c("TFAP2A-OE","TFAP2A-OE",
                              "GFP-OE","GFP-OE","ES_D0","ES_D0","NC_D5","NC_D5"),
                       levels = c("TFAP2A-OE","GFP-OE","ES_D0","NC_D5"))
g <- ggplot(pca_mat, aes(x=PC1, y=PC2, color=condition)) + 
  geom_point(size = 4, alpha = 1) + 
  scale_color_npg() +
  labs(x = "PC1 64%" ,y = "PC2 22%") +
  theme(text=element_text(color="black", size = 10, face = "bold"),
        line=element_line(color="black", size = 2),
        axis.ticks = element_line(color = "black", size = 1),
        panel.background = element_blank(),
        panel.border = element_rect(fill=NA, color = "black", size = 2),
        axis.text = element_text(color="black", size = 10, face = "bold"),
        legend.key = element_blank()
  )

g

ggsave("Exports/TFAP2A_OE_ATAC_PCA_D0-D5.png", width = 6, height = 4)
ggsave("Exports/TFAP2A_OE_ATAC_PCA_D0-D5.pdf", width = 6, height = 4)
```

Let's look at the accessibility regions enriched from TFAP2A-OE in D0 and D5 WT ATAC.

```{r}
# Top Enriched Peaks in TFAP2A-OE
TFAP2A_Enriched

library(DESeq2)
OE_ATAC_SE <- dba(OE_dba, bSummarizedExperiment = T)
OE_ATAC_SE@assays@data$scores <- OE_ATAC_SE@assays@data$Reads

OE_ATAC_dds <- DESeqDataSet(OE_ATAC_SE, design = ~ Condition)
OE_ATAC_dds <- DESeq(OE_ATAC_dds)
OE_ATAC_res <- results(OE_ATAC_dds)
OE_ATAC_rld <- rlog(OE_ATAC_dds, blind = T)
OE_ATAC_rld_df <- as.data.frame(assay(OE_ATAC_rld))

mod_rownames <- as.data.frame(OE_ATAC_SE@rowRanges)
mod_rownames$PeakID <- paste0(mod_rownames$seqnames, "-", mod_rownames$start, "-", mod_rownames$end)
OE_ATAC_rld_df$PeakID <- mod_rownames$PeakID

atac_rlog_counts_melted <- melt(OE_ATAC_rld_df)
atac_rlog_counts_melted$Group <- ifelse(atac_rlog_counts_melted$PeakID %in% TFAP2A_Enriched$PeakID,
                                   "TFAP2A-OE_Enriched","Other")

atac_rlog_counts_melted$Factor <- as.factor(atac_rlog_counts_melted$variable)

levels(atac_rlog_counts_melted$Factor) <- c("TFAP2A-OE","TFAP2A-OE","GFP-OE","GFP-OE","D0","D0","D5","D5")
atac_rlog_counts_melted$Factor <- relevel(atac_rlog_counts_melted$Factor, ref = "GFP-OE")
atac_rlog_counts_melted$Factor <- relevel(atac_rlog_counts_melted$Factor, ref = "TFAP2A-OE")


my_comparisons <- list( c("D0", "D5"), c("D0", "GFP-OE"), c("TFAP2A-OE","D5"), c("TFAP2A-OE","GFP-OE")
                        , c("D5","GFP-OE"))

ggplot(atac_rlog_counts_melted, aes(Factor, value)) + geom_boxplot(aes(fill = Factor)) + theme(axis.text.x = element_blank()) + stat_compare_means(comparisons = my_comparisons, method.args = list(alternative = "two.sided")) + scale_fill_brewer(palette = "Set1") + ggtitle("Enrichment of TFAP2A-OE induced gene expression as compared to wild type.") + xlab("ATAC-Seq Library") + ylab("Rlog Counts")s

# Instead, let's look at how these regions change in profile via deeptools.
```
```{bash}
computeMatrix reference-point -bl /data/Austin/workdir/genome/hg38/blacklist/UCSC_hg38_blacklist.bed -S /data/Austin/workdir/NC_Timecourse_Data/H1_TFAP2A_OE2_ATAC/BW/12793_10099_147159_HGHM3BGXK_H1_Ap2A_ATAC2_TCCTGAGC_toGRCh38_sorted_nodups.bw /data/Austin/workdir/NC_Timecourse_Data/H1_TFAP2A_OE2_ATAC/BW/12793_10099_147157_HGHM3BGXK_H1_GFP_ATAC2_CGTACTAG_toGRCh38_sorted_nodups.bw /data/Austin/workdir/NC_Timecourse_Data/Human_ESC_ATAC/BW/ATAC_ES_D0_1_toGRCh38_sorted_nodups.bw /data/Austin/workdir/NC_Timecourse_Data/Human_ESC_ATAC/BW/ATAC_NC_D5_2_toGRCh38_sorted_nodups.bw  -R ~/local_git/NC_Timecourse/ATAC-Seq/Human_ESC_iNC/Exports/TFAP2A_OE_Enriched_Peaks.bed -b 1000 -a 1000 -bs 5 -o TFAP2A_OE_Enriched_Peaks.mat.gz -p 62 --missingDataAsZero --referencePoint center

plotHeatmap -m TFAP2A_OE_Enriched_Peaks.mat.gz -out TFAP2A_OE_Enriched_Peaks.eps --colorList "#0584bc","#ffffff","#e45649" --dpi 600 --heatmapHeight 8

plotHeatmap -m TFAP2A_OE_Enriched_Peaks.mat.gz -out TFAP2A_OE_Enriched_Peaks.pdf --colorList "#0584bc","#ffffff","#e45649" --dpi 600 --heatmapHeight 8

gs -dSAFER -dEPSCrop -r300 -sDEVICE=pngalpha -o TFAP2A_OE_Enriched_Peaks.png TFAP2A_OE_Enriched_Peaks.eps

# Looks good.
```

Run HOMER on Enriched Peaks
```{bash}
findMotifsGenome.pl TFAP2A_OE_Enriched_Peaks.bed /data/Austin/workdir/genome/hg38/dna/hg38.fa TFAP2A_OE2_Enriched_HOMER -p 62
```

Looking at footprinting

## Running HINT-ATAC

```{bash}
cd /data/Austin/workdir/NC_Timecourse_Data/H1_TFAP2A_OE2_ATAC/
mkdir HINT

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT  --output-prefix=TFAP2A_1_HINT ./BAM/12793_10099_147158_HGHM3BGXK_H1_Ap2A_ATAC1_AGGCAGAA_toGRCh38_sorted_nodups.bam ./Peaks/12793_10099_147158_HGHM3BGXK_H1_Ap2A_ATAC1_AGGCAGAA_toGRCh38_sorted_nodups.bam_peaks.narrowPeak

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT  --output-prefix=TFAP2A_2_HINT ./BAM/12793_10099_147159_HGHM3BGXK_H1_Ap2A_ATAC2_TCCTGAGC_toGRCh38_sorted_nodups.bam ./Peaks/12793_10099_147159_HGHM3BGXK_H1_Ap2A_ATAC2_TCCTGAGC_toGRCh38_sorted_nodups.bam_peaks.narrowPeak

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT  --output-prefix=GFP_1_HINT ./BAM/12793_10099_147156_HGHM3BGXK_H1_GFP_ATAC1_TAAGGCGA_toGRCh38_sorted_nodups.bam ./Peaks/12793_10099_147156_HGHM3BGXK_H1_GFP_ATAC1_TAAGGCGA_toGRCh38_sorted_nodups.bam_peaks.narrowPeak

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT  --output-prefix=GFP_2_HINT ./BAM/12793_10099_147157_HGHM3BGXK_H1_GFP_ATAC2_CGTACTAG_toGRCh38_sorted_nodups.bam ./Peaks/12793_10099_147157_HGHM3BGXK_H1_GFP_ATAC2_CGTACTAG_toGRCh38_sorted_nodups.bam_peaks.narrowPeak

# Compare outputs
rgt-motifanalysis matching --organism=hg38 --motif-dbs ~/rgtdata/motifs/JASPAR2020-Vert-NonRedundant --rmdup --input-files ./HINT/TFAP2A_1_HINT.bed ./HINT/TFAP2A_2_HINT.bed ./HINT/GFP_1_HINT.bed ./HINT/GFP_2_HINT.bed



```

Test TFAP2A-OE against GFP-OE

```{bash}
rgt-hint differential --organism=hg38 --bc --nc 62 \
--mpbs-files=./match/TFAP2A_1_HINT_mpbs.bed,./match/TFAP2A_2_HINT_mpbs.bed,./match/GFP_1_HINT_mpbs.bed,./match/GFP_2_HINT_mpbs.bed --conditions=TFAP2A_OE,TFAP2A_OE,GFP_OE,GFP_OE --reads-files=./BAM/12793_10099_147158_HGHM3BGXK_H1_Ap2A_ATAC1_AGGCAGAA_toGRCh38_sorted_nodups.bam,./BAM/12793_10099_147159_HGHM3BGXK_H1_Ap2A_ATAC2_TCCTGAGC_toGRCh38_sorted_nodups.bam,./BAM/12793_10099_147156_HGHM3BGXK_H1_GFP_ATAC1_TAAGGCGA_toGRCh38_sorted_nodups.bam,./BAM/12793_10099_147157_HGHM3BGXK_H1_GFP_ATAC2_CGTACTAG_toGRCh38_sorted_nodups.bam --output-location=TFAP2A_OE_VS_GFP_Enriched_Footprinting

```

This gave some good results, but let's focus our analysis on those peaks specifically enriched in TFAP2A-OE.

```{bash}
cd /data/Austin/workdir/NC_Timecourse_Data/H1_TFAP2A_OE2_ATAC/
mkdir HINT_TFAP2A_OE2_Enriched

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT_TFAP2A_OE2_Enriched  --output-prefix=TFAP2A_1_HINT_TFAP2A_OE2_Enriched ./BAM/12793_10099_147158_HGHM3BGXK_H1_Ap2A_ATAC1_AGGCAGAA_toGRCh38_sorted_nodups.bam ~/local_git/NC_Timecourse/ATAC-Seq/Human_ESC_iNC/Exports/TFAP2A_OE_Enriched_Peaks.bed

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT_TFAP2A_OE2_Enriched  --output-prefix=TFAP2A_2_HINT_TFAP2A_OE2_Enriched ./BAM/12793_10099_147159_HGHM3BGXK_H1_Ap2A_ATAC2_TCCTGAGC_toGRCh38_sorted_nodups.bam ~/local_git/NC_Timecourse/ATAC-Seq/Human_ESC_iNC/Exports/TFAP2A_OE_Enriched_Peaks.bed

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT_TFAP2A_OE2_Enriched  --output-prefix=GFP_1_HINT_TFAP2A_OE2_Enriched ./BAM/12793_10099_147156_HGHM3BGXK_H1_GFP_ATAC1_TAAGGCGA_toGRCh38_sorted_nodups.bam ~/local_git/NC_Timecourse/ATAC-Seq/Human_ESC_iNC/Exports/TFAP2A_OE_Enriched_Peaks.bed

rgt-hint footprinting --atac-seq --paired-end --organism=hg38 --output-location=./HINT_TFAP2A_OE2_Enriched  --output-prefix=GFP_2_HINT_TFAP2A_OE2_Enriched ./BAM/12793_10099_147157_HGHM3BGXK_H1_GFP_ATAC2_CGTACTAG_toGRCh38_sorted_nodups.bam ~/local_git/NC_Timecourse/ATAC-Seq/Human_ESC_iNC/Exports/TFAP2A_OE_Enriched_Peaks.bed

rgt-motifanalysis matching --organism=hg38 --motif-dbs ~/rgtdata/motifs/JASPAR2020-Vert-NonRedundant --rmdup --input-files  ./HINT_TFAP2A_OE2_Enriched/TFAP2A_1_HINT_TFAP2A_OE2_Enriched.bed ./HINT_TFAP2A_OE2_Enriched/TFAP2A_2_HINT_TFAP2A_OE2_Enriched.bed ./HINT_TFAP2A_OE2_Enriched/GFP_1_HINT_TFAP2A_OE2_Enriched.bed ./HINT_TFAP2A_OE2_Enriched/GFP_2_HINT_TFAP2A_OE2_Enriched.bed 

rgt-hint differential --organism=hg38 --bc --nc 62 \
--mpbs-files=./match/TFAP2A_1_HINT_TFAP2A_OE2_Enriched_mpbs.bed,./match/TFAP2A_2_HINT_TFAP2A_OE2_Enriched_mpbs.bed,./match/GFP_1_HINT_TFAP2A_OE2_Enriched_mpbs.bed,./match/GFP_2_HINT_TFAP2A_OE2_Enriched_mpbs.bed --conditions=TFAP2A_OE,TFAP2A_OE,GFP_OE,GFP_OE --reads-files=./BAM/12793_10099_147158_HGHM3BGXK_H1_Ap2A_ATAC1_AGGCAGAA_toGRCh38_sorted_nodups.bam,./BAM/12793_10099_147159_HGHM3BGXK_H1_Ap2A_ATAC2_TCCTGAGC_toGRCh38_sorted_nodups.bam,./BAM/12793_10099_147156_HGHM3BGXK_H1_GFP_ATAC1_TAAGGCGA_toGRCh38_sorted_nodups.bam,./BAM/12793_10099_147157_HGHM3BGXK_H1_GFP_ATAC2_CGTACTAG_toGRCh38_sorted_nodups.bam --output-location=TFAP2A_OE_VS_GFP_Enriched_Footprinting
```

```{r}
differential_statistics <- read_delim("/data/Austin/workdir/NC_Timecourse_Data/H1_TFAP2A_OE2_ATAC/TFAP2A_OE_VS_GFP_Enriched_Footprinting/differential_statistics.txt", 
    delim = "\t", escape_double = FALSE, 
    trim_ws = TRUE)
```

# SOX2 CUT&RUN Analysis in TFAP2A vs GFP OE
In the 1_Sox2_Dimer_Timecourse doc

