---
title: "R Notebook"
output: html_notebook
---

I want to check on the conservation scores of each PeakID to supplement the correlation between the peak and gene.

```{r}
#### Creating the appropriate score objects -------------------
library(rtracklayer)
library(stringr)
#chrs is a character vector of all primary contigs (excluding MT)
chrs <- c("W","Z","1","2","3","4","5","6","7","8","9","10","11",
          "12","13","14","15","16","17","18","19","20","21","22","23",
          "24","25","26","27","28","30","31","32","33")
ucsc_phastcons <- import("/data/Austin/workdir/genome/galGal6/galGal6_conservation/galGal6.phastCons77way.bw", format = "BigWig")
seqlevelsStyle(ucsc_phastcons) <- "Ensembl"
seql_1 <- seqlevels(ucsc_phastcons)
seql_1[208] <- "Z"
seql_1[205] <- "W"
seqlevels(ucsc_phastcons) <- seql_1
ensembl_phastcons_filt <- keepSeqlevels(ucsc_phastcons, value = chrs, pruning.mode = "coarse")
rm(ucsc_phastcons)






library(rtracklayer)
library(stringr)
library(ChIPseeker)
options(scipen=999)

All_Genes_Tests <- readRDS(file = "Exports/All_Genes_Tests.RDS")
ATAC_all_peaks = readPeakFile("Imports/all_ATAC_400bp_fixed.bed")
ATAC_all_peaks <- as.data.frame(ATAC_all_peaks)
ATAC_all_peaks$PeakID <- paste0(ATAC_all_peaks$seqnames, "-", ATAC_all_peaks$start -1 , "-", ATAC_all_peaks$end)
summary(All_Genes_Tests$PeakID %in% ATAC_all_peaks$PeakID)
temp <- All_Genes_Tests[!All_Genes_Tests$PeakID %in% ATAC_all_peaks$PeakID,]
split <- str_split(unique(All_Genes_Tests$PeakID), pattern = "-", simplify = T)
PeakID.gr <- makeGRangesFromDataFrame(data.frame(seqnames = split[,1], start = as.numeric(split[,2]), end = as.numeric(split[,3])))

Window25bp.gr <- readRDS(file = "Imports/Phastcons_77vert_25bpWindows.gr.RDS")
list_of_gr <- split(PeakID.gr, ceiling(seq_along(PeakID.gr)/1000))

res_df <- data.frame()

chunk1 <-  1:(length(list_of_gr)/3)
chunk2 <- 74:146
chunk3 <- 147:217

#87 seems to fail for some reason.

for (gr in c(87)){
  message(paste0("Working on ", gr, " of ", length(list_of_gr)))
  temp <- do.call(c, list_of_gr[gr])[[1]]
  hits <- findOverlaps(temp, Window25bp.gr)
  gc()
  agg <- aggregate(Window25bp.gr, hits, score=mean(score))
  temp$score[countQueryHits(hits) > 0L] <- agg$score
  gc()
  res_df <- rbind(res_df, as.data.frame(temp))
}

saveRDS(res_df, file = "chunk2.RDS")

```

Unused stuff
```{r}
save(ensembl_phastcons_filt, file = "ensembl_phastcons_filt.gr")
load("ensembl_phastcons_filt.gr")

#First assign a starting value to each location based on chr.
#To get this, use the number of runs of each chr.
run_lengths = data.frame(chr = ensembl_phastcons_filt@seqnames@values, runs = ensembl_phastcons_filt@seqnames@lengths)
run_lengths = transform(run_lengths, new_start = ave(runs, FUN = cumsum)-194562278)

seq_key <- split(run_lengths[, 3], as.character(run_lengths[, 1]))

#So chr1 = 0
# chr2 = ...
# And then add this to the start and end ranges of WINDOW.
window(ensembl_phastcons_filt, start = 194562275, end = 194562280)
x="1_2000_5000"

get_conservation_2("1_2000_5000")
```

