---
title: "Generating initial ATAC_counts"
output: html_notebook
---
## ATAC-Seq Step 1: QC
Our first step in evaluating ATAC-Seq data is by running a variety of QC measures. We will operate on the dba object from diffbind to store the reads (bams) from our samples. For simplicity of github, we will just be working with the final object, but the bams for our Human data can be found at CREST2 `/data/Austin/workdir/NC_Timecourse_Data/Human_ESC_ATAC/BAM`
```{r Libraries-1, echo=FALSE, warning=FALSE}
library(GenomicRanges)
library(DiffBind)
library(BiocParallel)
library(rtracklayer)
library(tibble)
library(dplyr)
library(GenomicFeatures)
library(ChIPseeker)
BiocParallel::multicoreWorkers()
library(ggplot2)
library(ggrepel)
library(pheatmap)
library(ggsci)
library(Rsamtools)
library(GenomicAlignments)
library(ChIPQC)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
TxDb_hg38 <- TxDb.Hsapiens.UCSC.hg38.knownGene
library(AnnotationHub)
ah <- AnnotationHub()
qr <- query(ah, c("EnsDb", "Homo sapiens"))
ensembl_human_annotation <- qr[[14]]
```
Let's generate some QC metrics for these BAMs.
We will show one of the results as a demonstration, but this has been done for all libraries.
```{r, message=FALSE, collapse=TRUE}
#create a list of files that end with .noDupsSort.bam ($ specifies that the file ends there.)
bamFiles <- list.files(path = "/data/Austin/workdir/NC_Timecourse_Data/Human_ESC_ATAC/BAM",
                       pattern = "_sorted_nodups.bam$", full.names = T)
peakFiles <- list.files(path = "/data/Austin/workdir/NC_Timecourse_Data/Human_ESC_ATAC/Peaks",
                        pattern = "_sorted_nodups.bam_peaks.narrowPeak$", full.names = T) 


Perform_ATAC_QC <- function(bam = "", peaks = "") {
  reads1 <-
    readGAlignmentPairs(bam, param = ScanBamParam(
      mapqFilter = 1,
      flag = scanBamFlag(isPaired = TRUE, isProperPair = TRUE),
      what = c("qname", "mapq", "isize")
    ))
  idx_tb <- idxstatsBam(bam)
  idx_tb_filt <- idx_tb[0:25,]
  read_dist <- idx_tb_filt %>% ggplot(aes(seqnames, mapped, fill = seqnames)) +
    geom_bar(stat = "identity") + coord_flip()
  percent.mito <-(idx_tb_filt$mapped[25] / sum(idx_tb_filt$mapped)) * 100
  #getting insert sizes
  atacReads_read1 <- GenomicAlignments::first(reads1)
  insertSizes <- abs(elementMetadata(atacReads_read1)$isize)
  fragLenPlot <- table(insertSizes) %>%
    data.frame %>%
    rename(InsertSize = insertSizes,Count = Freq) %>%
    mutate(InsertSize = as.numeric(as.vector(InsertSize)),
           Count =as.numeric(as.vector(Count))) %>% ggplot(aes(x = InsertSize, y = Count)) +
    geom_line() + xlim(c(0, 800)) + scale_y_continuous(trans='log2', name = "Count Log2") + theme_bw()
  openRegionPeaks <- import(peaks, format = "narrowPeak")
  qcRes <-ChIPQCsample(bam, peaks = openRegionPeaks, chromosomes = "chr2",
       blacklist ="/data/Austin/workdir/genome/hg38/blacklist/UCSC_hg38_blacklist.bed")
  MacsCalls_1_filteredAnno <- annotatePeak(qcRes, TxDb = TxDb_hg38)
  output <-list(idx_tb,
                read_dist,
                percent.mito,
                fragLenPlot,
                MacsCalls_1_filteredAnno)
  return(output)
}

#Perform QC on all bam files. Easiest is to run one per bam, but could do for file in bamFiles and return those results in one big list.
res_list <- list()
for (i in seq(length(bamFiles))){
  name <- list.files(path = "/data/Austin/workdir/NC_Timecourse_Data/Human_ESC_ATAC/BAM",
                       pattern = "_sorted_nodups.bam$")[i]
  name <- strsplit(x = name, split = ".bam")[[1]]
  res_list[[name]] <- Perform_ATAC_QC(bam = bamFiles[i], peaks = peakFiles[i])
}

# They look like they have very low mito and decent signal.
# Let's check the bigwigs
# They all looks prety great!
```

# ESC vs D5 NC Analysis
```{r creating db object, echo=TRUE, eval=FALSE}
NC_dba <- dba(sampleSheet="ESC_SampleSheet_NC.csv")
NC_dba <- dba.count(NC_dba)
# All ESC vs NC
#NC_dba <- dba.contrast(NC_dba, NC_dba$masks$ES, NC_dba$masks$NC)
# ESC vs D3
#NC_dba <- dba.contrast(NC_dba, NC_dba$masks$ES, NC_dba$masks$D3)
# ESC vs D5 
# Negative fold change is enriched in D5, positive is enriched in ESC
NC_dba <- dba.contrast(NC_dba, NC_dba$masks$ES, NC_dba$masks$D5)
# D3 vs D5
#NC_dba <- dba.contrast(NC_dba, NC_dba$masks$D3, NC_dba$masks$D5)
NC_dba <- dba.analyze(NC_dba, bParallel = T)
NC_res <- dba.report(NC_dba)



# Make other PCA plot
dba.plotPCA(NC_dba)
meta_dataframe <- as.data.frame(NC_dba$class)
pca <- dba.plotPCA(NC_dba)
x <- pca$panel.args[[1]]$x
y <- pca$panel.args[[1]]$y
pca_mat <- as.data.frame(matrix(c(x,y), nrow = 6, ncol = 2, byrow = F))
colnames(pca_mat) <- c("PC1","PC2")
rownames(pca_mat) <- colnames(meta_dataframe)

pca_mat$timepoint <- factor(c("ES","ES","NC3","NC3",
                       "NC5","NC5"),
                       levels = c("ES","NC3","NC5"))
g <- ggplot(pca_mat, aes(x=PC1, y=PC2, color=timepoint)) + 
  geom_point(size = 4, alpha = 1) + 
  scale_color_npg() +
  labs(x = "PC1 64%" ,y = "PC2 22%") +
  theme(text=element_text(color="black", size = 10, face = "bold"),
        line=element_line(color="black", size = 2),
        axis.ticks = element_line(color = "black", size = 1),
        panel.background = element_blank(),
        panel.border = element_rect(fill=NA, color = "black", size = 2),
        axis.text = element_text(color="black", size = 10, face = "bold"),
        legend.key = element_blank()
  )

g

ggsave('./Exports/Human_ESC-to-iNC-PCA.eps', units = 'in', height = 4, width = 6, device = "eps")
```

## Exporting enriched peaksets.
```{r}
NC_res_ann <- annotatePeak(NC_res, TxDb = TxDb_hg38)

library(qdapTools)
NC_res_ann_df <- as.data.frame(NC_res_ann)
txs <- as.data.frame(transcripts(ensembl_human_annotation))
genes <- as.data.frame(genes(ensembl_human_annotation))
txs_lookup <- txs[,c("tx_id_version","gene_id")]
genes_lookup <- genes[,c("gene_id","gene_name")]
NC_res_ann_df$ENSEMBL <- lookup(NC_res_ann_df$transcriptId, txs_lookup)
NC_res_ann_df$Symbol <- lookup(NC_res_ann_df$ENSEMBL, genes_lookup)
NC_res_ann_df$peakID <- paste0(NC_res_ann_df$seqnames, "_",NC_res_ann_df$start,"_",NC_res_ann_df$end)


D5_NC_Enriched_5k <- top_n(NC_res_ann_df, n = 5000, desc(Fold)) 
D5_ESC_Enriched_5k <- top_n(NC_res_ann_df, n = 5000, Fold) 

D5_NC_Enriched_5k.gr <- makeGRangesFromDataFrame(D5_NC_Enriched_5k)
D5_ESC_Enriched_5k.gr <- makeGRangesFromDataFrame(D5_ESC_Enriched_5k)

export.bed(D5_NC_Enriched_5k.gr, con = "Exports/D5_NC_Enriched_5k.bed")
export.bed(D5_ESC_Enriched_5k.gr, con = "Exports/D5_ESC_Enriched_5k.bed")
```

## Run homer on ESC and NC enriched peaks
```{bash}
findMotifsGenome.pl D5_ESC_Enriched_5k.bed hg38 D5_ESC_Enriched_5k_HOMER -p 31

findMotifsGenome.pl D5_NC_Enriched_5k.bed hg38 D5_NC_Enriched_5k_HOMER -p 31
```

Results of this show strong Oct4, Sox2, and Oct4::Sox2 enrichment in ESC, good.
And a lot of GATA2 results in D5 NC.

# ESC vs D3 NC Analysis
```{r}
# ESC vs D3
NC_dba$contrasts=NULL
# ESC Enriched positive LogFC, D3 enriched Negative.
NC_dba <- dba.contrast(NC_dba, NC_dba$masks$ES, NC_dba$masks$D3)
NC_dba <- dba.analyze(NC_dba, bParallel = T)
NC_res <- dba.report(NC_dba)

NC_res_ann <- annotatePeak(NC_res, TxDb = TxDb_hg38)

library(qdapTools)
NC_res_ann_df <- as.data.frame(NC_res_ann)
txs <- as.data.frame(transcripts(ensembl_human_annotation))
genes <- as.data.frame(genes(ensembl_human_annotation))
txs_lookup <- txs[,c("tx_id_version","gene_id")]
genes_lookup <- genes[,c("gene_id","gene_name")]
NC_res_ann_df$ENSEMBL <- lookup(NC_res_ann_df$transcriptId, txs_lookup)
NC_res_ann_df$Symbol <- lookup(NC_res_ann_df$ENSEMBL, genes_lookup)
NC_res_ann_df$peakID <- paste0(NC_res_ann_df$seqnames, "_",NC_res_ann_df$start,"_",NC_res_ann_df$end)

D3_NC_Enriched_5k <- top_n(NC_res_ann_df, n = 5000, desc(Fold)) 
D3_ESC_Enriched_5k <- top_n(NC_res_ann_df, n = 5000, Fold) 

D3_NC_Enriched_5k.gr <- makeGRangesFromDataFrame(D3_NC_Enriched_5k)
D3_ESC_Enriched_5k.gr <- makeGRangesFromDataFrame(D3_ESC_Enriched_5k)

export.bed(D3_NC_Enriched_5k.gr, con = "Exports/D3_NC_Enriched_5k.bed")
export.bed(D3_ESC_Enriched_5k.gr, con = "Exports/D3_ESC_Enriched_5k.bed")
```

## Run homer on ESC and NC enriched peaks
```{bash}
findMotifsGenome.pl D3_NC_Enriched_5k.bed hg38 D3_NC_Enriched_5k_HOMER -p 31

findMotifsGenome.pl D3_ESC_Enriched_5k.bed hg38 D3_ESC_Enriched_5k_HOMER -p 31
```

# D3 vs D5 NC Analysis
```{r}
# ESC vs D3
NC_dba$contrasts=NULL
# D3 Enriched positive LogFC, D5 enriched Negative.
NC_dba <- dba.contrast(NC_dba, NC_dba$masks$D3, NC_dba$masks$D5)
NC_dba <- dba.analyze(NC_dba, bParallel = T)
NC_res <- dba.report(NC_dba)

NC_res_ann <- annotatePeak(NC_res, TxDb = TxDb_hg38)

library(qdapTools)
NC_res_ann_df <- as.data.frame(NC_res_ann)
txs <- as.data.frame(transcripts(ensembl_human_annotation))
genes <- as.data.frame(genes(ensembl_human_annotation))
txs_lookup <- txs[,c("tx_id_version","gene_id")]
genes_lookup <- genes[,c("gene_id","gene_name")]
NC_res_ann_df$ENSEMBL <- lookup(NC_res_ann_df$transcriptId, txs_lookup)
NC_res_ann_df$Symbol <- lookup(NC_res_ann_df$ENSEMBL, genes_lookup)
NC_res_ann_df$peakID <- paste0(NC_res_ann_df$seqnames, "_",NC_res_ann_df$start,"_",NC_res_ann_df$end)

D3_vs_D5_Enriched_5k <- top_n(NC_res_ann_df, n = 5000, desc(Fold)) 
D5_vs_D3_Enriched_5k <- top_n(NC_res_ann_df, n = 5000, Fold) 

D3_vs_D5_Enriched_5k.gr <- makeGRangesFromDataFrame(D3_vs_D5_Enriched_5k)
D5_vs_D3_Enriched_5k.gr <- makeGRangesFromDataFrame(D5_vs_D3_Enriched_5k)

export.bed(D3_vs_D5_Enriched_5k.gr, con = "Exports/D3_vs_D5_Enriched_5k.bed")
export.bed(D5_vs_D3_Enriched_5k.gr, con = "Exports/D5_vs_D3_Enriched_5k.bed")
```

## Run homer on ESC and NC enriched peaks
```{bash}
findMotifsGenome.pl D3_NC_Enriched_5k.bed hg38 D3_NC_Enriched_5k_HOMER -p 31

findMotifsGenome.pl D3_ESC_Enriched_5k.bed hg38 D3_ESC_Enriched_5k_HOMER -p 31
```

## Normalized counts via DESeq2
While we have used diffbind to generate peaksets, we do not have an easy way to generate normalized counts directly from the diffbind object. Therefore, we must transform it into a DESeq2 object to generate a set of normalized counts for additional analysis.
```{r DESeq2, echo=TRUE}
# Running DeSeq to get Rlog counts
library(DESeq2)
summarized_experiment2 <- dba(NC_dba, bSummarizedExperiment = T)
summarized_experiment2@assays@data$scores <- summarized_experiment2@assays@data$Reads
#order of counts not correct.
dds_experiment2 <- DESeqDataSet(summarized_experiment2, design = ~ Condition)
dds_experiment2 <- DESeq(dds_experiment2)
dds_results <- results(dds_experiment2, contrast = c("Condition","D0","D5"))

rownames(dds_results) <- paste0(NC_dba$peaks[[1]]$Chr,"_",NC_dba$peaks[[1]]$Start,"_",NC_dba$peaks[[1]]$End)
rld <- rlog(dds_experiment2, blind = FALSE)

rownames(assay(rld)) <- paste0(NC_dba$peaks[[1]]$Chr,"_",NC_dba$peaks[[1]]$Start,"_",NC_dba$peaks[[1]]$End)
rld_df <- data.frame(row.names = paste0(NC_dba$peaks[[1]]$Chr,"_",NC_dba$peaks[[1]]$Start,"_",NC_dba$peaks[[1]]$End),assay(rld))

write.table(rld_df, file = "./Exports/Rlog_counts_All_Human_ESC-NC_ATAC_Peaks", quote = F, sep = "\t", row.names = T)

library(pheatmap)

rld_df_avg <- data.frame(ES_D0 = rowMeans(rld_df[c('ATAC_ES_D0_1', 'ATAC_ES_D0_2')], na.rm=TRUE),
                         NC_D3 = rowMeans(rld_df[c('ATAC_NC_D3_1', 'ATAC_NC_D3_2')], na.rm=TRUE),
                         NC_D5 = rowMeans(rld_df[c('ATAC_NC_D5_1', 'ATAC_NC_D5_2')], na.rm=TRUE))

write.table(rld_df_avg, file = "./Exports/Rlog_counts_Avg_All_Human_ESC-NC_ATAC_Peaks", quote = F, sep = "\t", row.names = T)

deseq2_peakset <- as.data.frame(dds_results) %>% dplyr::filter(padj < 1e-5)
dds_results2 <- results(dds_experiment2, contrast = c("Condition","D3","D5"))
deseq2_peakset2 <- as.data.frame(dds_results2) %>% dplyr::filter(padj < 1e-5)
dds_results3 <- results(dds_experiment2, contrast = c("Condition","D0","D3"))
deseq2_peakset3 <- as.data.frame(dds_results3) %>% dplyr::filter(padj < 1e-5)

unique_peaks <- unique(c(rownames(deseq2_peakset),
         rownames(deseq2_peakset2),
         rownames(deseq2_peakset3)))

rld_df_subset <- subset(rld_df, rownames(rld_df) %in% unique_peaks)

ATAC_clusters = pheatmap(rld_df_subset,
                         cluster_cols = F, scale = "row",
                         show_rownames = F,show_colnames = T, cluster_rows = T, cutree_rows = 4,
                         color=colorRampPalette(colors=c("#0797B3","#FFFFFF","#4FA14E"))(250))

#So it renders properly in html doc
pheatmap(rld_df_subset,
                         cluster_cols = F, scale = "row",
                         show_rownames = F,show_colnames = T,
                         color=colorRampPalette(colors=c("#0797B3","#FFFFFF","#4FA14E"))(250))

#Re-order original data (genes) to match ordering in heatmap (top-to-bottom)
rownames(rld_df_subset[ATAC_clusters$tree_row[["order"]],])
#6 groups
clusters_4 <- data.frame(
  peak = names(sort(cutree(ATAC_clusters$tree_row, k=4))),
  cluster = as.character(sort(cutree(ATAC_clusters$tree_row, k=4))))

# Cluster 1 is only D5
# Cluster 2 is only ESC
# Cluster 3 is D3-D5 enriched
# Cluster 4 is ESC-D3 enriched
row_annotation <- data.frame(cluster = clusters_4$cluster, row.names = clusters_4$peak)
pheatmap(rld_df_subset, annotation_row = row_annotation, 
                         cluster_cols = F, scale = "row",
                         show_rownames = F,show_colnames = T, cluster_rows = T, cutree_rows = 4,
                         color=colorRampPalette(colors=c("#0797B3","#FFFFFF","#4FA14E"))(250),
         filename = "./Exports/Rlog_Counts_Human_DeSeq2_ATAC_peaks.pdf", width = 6, height = 5)
library(stringr)
clusters_bed <- clusters_4 %>%
  tidyr::separate(col = peak, sep = "_", into = c("chr","start","end"))
clusters_bed$start <- as.numeric(clusters_bed$start)
clusters_bed$end <- as.numeric(clusters_bed$end)

clusters_bed <- as.data.frame(clusters_bed[!is.na(clusters_bed),])

cluster_1 <- clusters_bed %>% dplyr::filter(cluster == 1) %>% dplyr::select(everything(), -cluster)
write.table(cluster_1, file = "./Exports/DeSeq2_ATAC_Cluster_1.bed",
            sep = "\t",row.names = F, col.names = F, quote = F)

cluster_2 <- clusters_bed %>% dplyr::filter(cluster == 2) %>% dplyr::select(everything(), -cluster)
write.table(cluster_2, file = "./Exports/DeSeq2_ATAC_Cluster_2.bed",
            sep = "\t",row.names = F, col.names = F, quote = F)

cluster_3 <- clusters_bed %>% dplyr::filter(cluster == 3) %>% dplyr::select(everything(), -cluster)
write.table(cluster_3, file = "./Exports/DeSeq2_ATAC_Cluster_3.bed",
            sep = "\t",row.names = F, col.names = F, quote = F)

cluster_4 <- clusters_bed %>% dplyr::filter(cluster == 4) %>% dplyr::select(everything(), -cluster)
write.table(cluster_4, file = "./Exports/DeSeq2_ATAC_Cluster_4.bed",
            sep = "\t",row.names = F, col.names = F, quote = F)
```

```{bash eval=FALSE, include=FALSE}
findMotifsGenome.pl DeSeq2_ATAC_Cluster_1.bed hg38 DeSeq2_ATAC_Cluster_1_HOMER -p 62
findMotifsGenome.pl DeSeq2_ATAC_Cluster_2.bed hg38 DeSeq2_ATAC_Cluster_2_HOMER -p 62
findMotifsGenome.pl DeSeq2_ATAC_Cluster_3.bed hg38 DeSeq2_ATAC_Cluster_3_HOMER -p 62
findMotifsGenome.pl DeSeq2_ATAC_Cluster_4.bed hg38 DeSeq2_ATAC_Cluster_4_HOMER -p 62
```
